### PR TITLE
Fix parsing in play.mvc.Result#charset()

### LINKED ISSUE #6280: Bug: default charset utf-8 not set the same way as read
### Play Version (2.5.x / etc)
Play 2.5.4

### API (Scala / Java / Neither / Both)
Java

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

OSX Yosemite

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)

### Library Dependencies

-

### Expected Behavior

1. Controller returns `Results.ok(someJSONNode);`
2. In my functional test: `assertThat(result.charset().get()).isEqualTo("utf-8");`
3. Assertion success

### Actual Behavior

1. Assertion fails with `java.util.NoSuchElementException: No value present`

That is because .get() tries to access an empty Optional value.

### Reproducible Test Case

-

I think that is because in StatusHeader.java:344 you set `Optional.of("application/json;charset=" + charset)`, but in Result.java you check against `if (h.contains("; charset=")) {`. There is a blank character in front of charset, which is missing in the set value.

This issue seems to be related to https://github.com/playframework/playframework/issues/5873

### PR DESCRIPTION
**Pull request by gmethvin**
Fixes #6280

### CODE REVIEW

**Comment by marcospereira** [comment]
Backport to 2.5.x: 8404fb1d775e437e930f5cc28e75049ac48f6ca2

**Comment by marcospereira** [comment]
Oops! I've decided to revert the manual backport because of #6288 (which is related to this issue, I suppose). Sorry.

@gmethvin, fell free to submit a PR against 2.5.x.

**Comment by gmethvin** [comment]
@marcospereira No big deal; it was just the imports on the test that had conflicts. Now the cherry pick is clean.

File: framework/src/play/src/test/scala/play/mvc/ResultSpec.scala [hunk 1]
```diff
@@ -39,5 +45,12 @@ object ResultSpec extends Specification {
       cookie.name() must be_==("name1")
       cookie.value() must be_==("value1")
     }
+
+    "get charset correctly" in {
+      val charset = StandardCharsets.ISO_8859_1.name()
+      val contentType = s"text/plain;charset=$charset"
```
Comment (ideadapt) [inline framework/src/play/src/test/scala/play/mvc/ResultSpec.scala]: Allthough you dont really test the regex white space magic with this test, I am happy the bug it self was fixed so quickly. Thanks!

**Pull request merged**